lib/modules/manager/gradle/artifacts.spec.ts

```diff
@@ -0,0 +1,392 @@
+      'gradle.lockfile': 'Current gradle.lockfile',
+    });
+    git.getRepoStatus.mockResolvedValue(
+      partial<StatusResult>({
+        modified: ['build.gradle', 'gradle.lockfile'],
+      })
+    );
+    fs.readLocalFile.mockResolvedValue('New gradle.lockfile');
+  });
+
+  it('aborts if no lockfile is found', async () => {
+    const execSnapshots = mockExecAll();
+    git.getFileList.mockResolvedValue(['build.gradle', 'settings.gradle']);
+
+    expect(
+      await updateArtifacts({
+        packageFileName: 'build.gradle',
+        updatedDeps: [],
+        newPackageFileContent: '',
+        config: {},
+      })
+    ).toBeNull();
+
+    expect(logger.logger.debug).toHaveBeenCalledWith(
+      'No Gradle dependency lockfiles found - skipping update'
+    );
+    expect(execSnapshots).toBeEmptyArray();
+  });
+
+  it('aborts if lock file exists but no gradle wrapper', async () => {
+    const execSnapshots = mockExecAll();
+    fs.findUpLocal.mockResolvedValue(null);
+
+    expect(
+      await updateArtifacts({
+        packageFileName: 'build.gradle',
+        updatedDeps: [],
+        newPackageFileContent: '',
+        config: {},
+      })
+    ).toBeNull();
+
+    expect(logger.logger.debug).toHaveBeenCalledWith(
+      'Found Gradle dependency lockfiles but no gradlew - aborting update'
+    );
+    expect(execSnapshots).toBeEmptyArray();
+  });
+
+  it('updates lock file', async () => {
+    const execSnapshots = mockExecAll();
+
+    const res = await updateArtifacts({
+      packageFileName: 'build.gradle',
+      updatedDeps: [
+        { depName: 'org.junit.jupiter:junit-jupiter-api' },
+        { depName: 'org.junit.jupiter:junit-jupiter-engine' },
+      ],
+      newPackageFileContent: '',
+      config: {},
+    });
+
+    expect(res).toEqual([
+      {
+        file: {
+          type: 'addition',
+          path: 'gradle.lockfile',
+          contents: 'New gradle.lockfile',
+        },
+      },
+    ]);
+    expect(execSnapshots).toMatchObject([
+      {
+        cmd: './gradlew --console=plain -q properties',
+        options: {
+          cwd: '/tmp/github/some/repo',
+        },
+      },
+      {
+        cmd: './gradlew --console=plain -q :dependencies --update-locks org.junit.jupiter:junit-jupiter-api,org.junit.jupiter:junit-jupiter-engine',
+        options: {
+          cwd: '/tmp/github/some/repo',
+        },
+      },
+    ]);
+  });
+
+  it('prefers packageName over depName if provided', async () => {
+    const execSnapshots = mockExecAll();
+
+    expect(
+      await updateArtifacts({
+        packageFileName: 'build.gradle',
+        updatedDeps: [
+          {
+            depType: 'plugin',
+            depName: 'org.springframework.boot',
+            packageName:
+              'org.springframework.boot:org.springframework.boot.gradle.plugin',
+          },
+        ],
+        newPackageFileContent: '',
+        config: {},
+      })
+    ).not.toBeNull();
```

Comment:
use toEqual instead, those tests caused wrong assumptions some te ago, as it could silently match artifact errors too